src/qibolab/result.py

```diff
@@ -61,7 +72,12 @@ def to_dict_probability(self, state=1):
         elif state == 0:
             return {"probability": self.ground_state_probability}
 
-    def to_dict(self, average=True):
+    def compute_average(self):
+        """Perform average over i and q"""
+        self.i = self.i.mean()
+        self.q = self.q.mean()
```

Comment:
What I don't like in this approach is that we lose information, if this function is called then we completely lose access to the individual values before the average. Also, it won't work well in cases where i and q are matrices eg. of shape (nshots, n_sweeped_values). In these cases we need to specify axis, otherwise the `mean` makes no sense.

Reply:
Definitely, I agree we should preserve the information, and drop only when the object goes out of scope.

Prefer returning a separate object, possibly with a dedicated type:
```py
def compute_average(self):
    """Perform average over i and q"""
    return ExecutionAverages(i=self.i.mean(), q=self.q.mean())
```
(the name is random, choose a suitable one)

Reply:
Ok, I'll give it a try. I imagine that in this case `ExecutionAverages` can be an inheritance of `ExecutionResults`, correct?

Reply:
Not sure this would work attribute-wise, but for sure you do not want to have the same method... so the answer is 'no' ^^

At the beginning, you should not need methods on it. Just make it struct-like (i.e. a method-less dataclass), and add the attributes you want to return. Just to give a type information about the content you are returning, and nothing else :)

Reply:
> Notice that with the `ExecRes` approach you don't need this function at all, since it boils down to a single `.mean()` call.

Are you sure about this? I think that you need to do the mean separately for `i` and `q` otherwise the following error is raised:

```python
import numpy as np
from qibolab.result import ExecutionResults
i = np.random.rand(5)
q = np.random.rand(5)

result = ExecutionResults.from_components(i,q)

result.array.mean()
```

```
---------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
Cell In[5], line 8
      4 q = np.random.rand(5)
      6 result = ExecutionResults.from_components(i,q)
----> 8 result.array.mean()

File ~/.conda/envs/qibocal/lib/python3.9/site-packages/numpy/core/_methods.py:181, in _mean(a, axis, dtype, out, keepdims, where)
    178         dtype = mu.dtype('f4')
    179         is_float16_result = True
--> 181 ret = umr_sum(arr, axis, dtype, out, keepdims, where=where)
    182 if isinstance(ret, mu.ndarray):
    183     with _no_nep50_warning():

UFuncTypeError: ufunc 'add' did not contain a loop with signature matching types (dtype((numpy.record, [('i', '<f8'), ('q', '<f8')])), dtype((numpy.record, [('i', '<f8'), ('q', '<f8')]))) -> None
```
Am I missing something?